bazel/cc_toolchains/clang_cc_toolchain_config.bzl

```diff
@@ -225,7 +225,7 @@ def _impl(ctx):
                 ],
                 flag_groups = [
                     flag_group(
-                        flags = ["-pie"],
+                        flags = ["-Wl,-pie"],
```

Comment:
While it appears today we don't rely on this, I think we want to keep `-pie` on Linux. It provides some other features (macro pre-defines) that to be consistent, we should keep using.

I suspect that the warning is actually correct on macOS and this is unused, so I think the correct fix here is to work out the platform conditions and make the Bazel here *conditional* and skip `-pie` on the macOS targets that we get a warning for (but not others). Does that make sense?